CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/scatter/Scatter3dArranger.java

```diff
@@ -0,0 +1,239 @@
+ * Copyright 2010-2020 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.arrangements.scatter;
+
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import au.gov.asd.tac.constellation.graph.GraphWriteMethods;
+import au.gov.asd.tac.constellation.graph.schema.analytic.attribute.objects.RawData;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.arrangements.Arranger;
+import au.gov.asd.tac.constellation.plugins.arrangements.GraphUtilities;
+import au.gov.asd.tac.constellation.utilities.color.ConstellationColor;
+import java.time.ZonedDateTime;
+import java.util.BitSet;
+
+/**
+ * Arrange the graph in a scatter3d.
+ *
+ * @author CrucisGamma
+ */
+public class Scatter3dArranger implements Arranger {
+
+    private final Scatter3dChoiceParameters params;
+
+    /**
+     * A scatter3d arrangement with default parameters.
+     */
+    public Scatter3dArranger() {
+        this(Scatter3dChoiceParameters.getDefaultParameters());
+    }
+
+    /**
+     * Construct new ArrangeInScatter3d instance.
+     *
+     * @param params Parameters for the arrangement.
+     */
+    public Scatter3dArranger(final Scatter3dChoiceParameters params) {
+        this.params = params;
+    }
+
+    @Override
+    public void setMaintainMean(final boolean b) {
+    }
+
+    @Override
+    public void arrange(final GraphWriteMethods wg) throws InterruptedException {
+
+        // Get/set the x,y,z attributes.
+        final int xAttr = VisualConcept.VertexAttribute.X.ensure(wg);
+        final int yAttr = VisualConcept.VertexAttribute.Y.ensure(wg);
+        final int zAttr = VisualConcept.VertexAttribute.Z.ensure(wg);
+
+        final String xDimension;
+        final String yDimension;
+        final String zDimension;
+        final boolean xLogarithmic;
+        final boolean yLogarithmic;
+        final boolean zLogarithmic;
+        if (params != null) {
+            xDimension = params.getXDimension();
+            yDimension = params.getYDimension();
+            zDimension = params.getZDimension();
+            xLogarithmic = params.getLogarithmicX();
+            yLogarithmic = params.getLogarithmicY();
+            zLogarithmic = params.getLogarithmicZ();
+        } else {
+            return;
+        }
+
+        final int xDim = wg.getAttribute(GraphElementType.VERTEX, xDimension);
```

Comment:
It took me a bit to understand what these `dimension` attributes were - maybe call them `xDimensionAttribute` or something to be clearer.